src/components/project/form-row.vue

```diff
@@ -0,0 +1,102 @@
+<!--
+Copyright 2017 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/getodk/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <tr class="form-row">
+    <!--todo: come back and figure out all the permissions-->
+    <td class="name">
+      <link-if-can :to="primaryFormPath(form)">
+        {{ form.nameOrId() }}
+      </link-if-can>
+    </td>
+    <td class="state-received">
+      {{ form.received }} recieved
+    </td>
+    <td class="state-edited">
+      {{ form.edited }} edited
+    </td>
+    <td class="state-has-issues">
+      {{ form.has_issues }} has issues
+    </td>
+    <td class="last-submission">
+      <div v-if="form.lastSubmission != null">
+        <router-link :to="submissionsPath">
+          <i18n :tag="false" path="common.lastSubmission">
+            <template #dateTime>
+              <date-time :iso="form.lastSubmission"/>
+            </template>
+          </i18n>
+        </router-link>
+      </div>
+    </td>
+    <td class="total-submissions">
+      <div v-if="form.publishedAt != null">
+        <router-link :to="submissionsPath">
+          <span>{{ $tcn('count.submission', form.submissions) }}</span>
+          <span class="icon-angle-right"></span>
+        </router-link>
+      </div>
+    </td>
+  </tr>
+</template>
+
+<script>
+import DateTime from '../date-time.vue';
+import EnketoFill from '../enketo/fill.vue';
+import EnketoPreview from '../enketo/preview.vue';
+import LinkIfCan from '../link-if-can.vue';
+import Form from '../../presenters/form';
+import routes from '../../mixins/routes';
+import { requestData } from '../../store/modules/request';
+
+export default {
+  name: 'ProjectFormRow',
+  components: { DateTime, EnketoFill, EnketoPreview, LinkIfCan },
+  mixins: [routes()],
+  props: {
+    form: {
+      type: Form,
+      required: true
+    },
+    columns: {
+      type: Set,
+      required: true
+    }
+  },
+  computed: {
+    // The component assumes that this data will exist when the component is
+    // created.
+    ...requestData(['project']),
+    submissionsPath() {
+      return this.formPath(
+        this.form.projectId,
+        this.form.xmlFormId,
+        this.form.publishedAt != null ? 'submissions' : 'draft/testing'
+      );
+    }
```

Comment:
The links for the review states will follow a similar pattern. For the review state of `null` (received), link to `submissions?reviewState=null`. For `edited`, `submissions?reviewState='edited'` (with the single quotes around `edited`). For `hasIssues`, `submissions?reviewState='hasIssues'`.

Not sure what to do if the form doesn't have a published version. The draft Testing page doesn't have filters, so I'm not sure it makes sense to link to a filtered view of that table. The counts will also be zero in that case, right?

Reply:
Actually, I think those single quotes should be encoded, so (for example) `submissions?reviewState=%27edited%27`.